### PR TITLE
Focus on the hero when he starts to move by the Move button press

### LINKED ISSUE #8030: Adventure map, screen focus jumps on a hero with a small delay
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

The screen shows a hero only a few moments after he started moving.

https://github.com/ihhub/fheroes2/assets/55348946/03f8d3a9-2be6-431e-915a-c90ce97c44b1

### Save file

[Path changes.zip](https://github.com/ihhub/fheroes2/files/13307465/Path.changes.zip)

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by Districh-ru**
- Fix #8030 (a regression after #8000).

### CODE REVIEW

File: src/fheroes2/gui/interface_events.cpp [hunk 1]
```diff
@@ -123,6 +115,23 @@ void Interface::AdventureMap::MoveHeroFromArrowKeys( Heroes & hero, const int di
     ShowPathOrStartMoveHero( &hero, dstIndex );
 }
 
+void Interface::AdventureMap::_startHeroMove( Heroes * hero )
```
Comment (ihhub) [inline src/fheroes2/gui/interface_events.cpp:118]: Please use a reference over the pointer. In both cases where we call this method we checked whether the pointer is **nullptr** so there is no need to do it again.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I left one suggestion here. Could you please take a look once you have time?

**Review by ihhub — APPROVED** [review]

**Review by Branikolog — APPROVED** [review]
Thank you, @Districh-ru !

**Comment by ihhub** [comment]
@Districh-ru , thank you so much for the fix!

**Pull request merged**